src/GraphQL.Tests/Types/IdGraphTypeTests.cs

```diff
@@ -10,9 +11,9 @@ public class IdGraphTypeTests
         private readonly IdGraphType _type = new IdGraphType();
 
         [Fact]
-        public void parse_value_null_to_null()
+        public void parse_value_null_throws()
         {
-            _type.ParseValue(null).ShouldBe(null);
+            Should.Throw<NullReferenceException>(() => _type.ParseValue(null));
```

Comment:
I think that previous behavior was correct. See my other comments about `ToAst`.

> the framework will never call

And... what? First, this may change. Secondly, it is a public method available to everyone. And it has pretty clear semantics. We ourselves wrote in the comments that
>  /// <returns> Internal scalar representation. Returning null indicates a failed conversion. </returns>

Reply:
Will be changed in #2333